### PR TITLE
add test: list rule with service group

### PR DESCRIPTION
**Pull request by cam-d12**
## PR summary
This PR adds CBR support for the new service_group_id parameter in rules.

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Current vs new behavior  
No current behavior should be modified. Adds the further control to be able to apply rules to service groups

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

File: test/integration/test_context_based_restrictions_v1.py [hunk 1]
```diff
@@ -417,6 +417,65 @@ def list_rules_with_invalid_account_id_parameter_error(self):
...
+        TestContextBasedRestrictionsV1.rule_id = rule['id']
+
+        list_rules_response = self.context_based_restrictions_service.list_rules(
+            service_group_id='IAM',
+            transaction_id=self.getTransactionID(),
+        )
+
+        assert list_rules_response.get_status_code() == 200
+        rule_list = list_rules_response.get_result()
+        assert rule_list is not None
```
Comment (soohoowoohoo) [inline test/integration/test_context_based_restrictions_v1.py:470]: I would expect the tests to assert that 1 rule was returned in the list and that it has the rule ID of the rule created above.

**Review by soohoowoohoo — CHANGES_REQUESTED** [review]

File: test/integration/test_context_based_restrictions_v1.py [hunk 2]
```diff
@@ -417,6 +417,67 @@ def list_rules_with_invalid_account_id_parameter_error(self):
...
+        assert rule_list['total_count'] == 1
```
Comment (soohoowoohoo) [inline test/integration/test_context_based_restrictions_v1.py:471]: does the python library support a `total_count` property? I know we have a `count` property but idk if there are any conversions being done somewhere in the python lib.

File: test/integration/test_context_based_restrictions_v1.py [hunk 3]
```diff
@@ -417,6 +417,67 @@ def list_rules_with_invalid_account_id_parameter_error(self):
...
+        assert rule['id'] == rule_list['first'].id
```
Comment (soohoowoohoo) [inline test/integration/test_context_based_restrictions_v1.py:472]: wouldn't we need to access the `rules` property first and don't python lists take indices like arrays?

**Review by soohoowoohoo — CHANGES_REQUESTED** [review]

**Comment by cam-d12** [comment]
@padamstx All integration tets passings but Travis CI build failing due to linting errors. Tried fixing locally and happy to try and other steps to resolve but otherwise I think this PR is good to go.

**Comment by padamstx** [comment]
Something must have happened with respect to the linter that is being used in our travis builds, because these linting errors look suspiciously like something that the generator would do.  I wonder if a new version of the linter is now complaining about these or something.

@pyrooka would you have any insight here?  All these linter errors seem to have blindsided us :)

**Comment by pyrooka** [comment]
Okay, I just finished some testing and @padamstx your intuition was good. A recently released Pylint has introduced this "issue". With <2.15.X the project builds fine, but with 2.15.0 and above it throws errors.
I've opened a separate [PR](https://github.com/IBM/platform-services-python-sdk/pull/169) with the fix, so after that get merged, we can update this PR.

**Review by soohoowoohoo — COMMENTED** [review]
LGTM

**Review by padamstx — APPROVED** [review]
LGTM

**Pull request merged**